engine-tests/src/test/java/org/terasology/documentation/ModuleMarkupScraper.java

```diff
@@ -0,0 +1,444 @@
+        if (gameplayModule != null) {
+            extensions.append("|  gameplayModule: | _").append(gameplayModule).append("_ |");
+            extensions.append(System.getProperty("line.separator"));
+        }
+        Boolean isAsset = (Boolean) metadata.getExtension(IS_ASSET.getKey(), Boolean.class);
+        if (isAsset != null) {
+            extensions.append("|  IS_ASSET: | _").append(isAsset).append("_ |");
+            extensions.append(System.getProperty("line.separator"));
+        }
+        Boolean isWorld = (Boolean) metadata.getExtension(IS_WORLD.getKey(), Boolean.class);
+        if (isAsset != null) {
+            extensions.append("|  IS_WORLD: | _").append(isWorld).append("_ |");
+            extensions.append(System.getProperty("line.separator"));
+        }
+        Boolean isLibrary = (Boolean) metadata.getExtension(StandardModuleExtension.IS_LIBRARY.getKey(), Boolean.class);
+        if (isAsset != null) {
+            extensions.append("|  IS_LIBRARY: | _").append(isLibrary).append("_ |");
+            extensions.append(System.getProperty("line.separator"));
+        }
+        Boolean isSpecial = (Boolean) metadata.getExtension(StandardModuleExtension.IS_SPECIAL.getKey(), Boolean.class);
+        if (isAsset != null) {
+            extensions.append("|  IS_SPECIAL: | _").append(isSpecial).append("_ |");
+            extensions.append(System.getProperty("line.separator"));
+        }
+        Boolean isAug = (Boolean) metadata.getExtension(StandardModuleExtension.IS_AUGMENTATION.getKey(), Boolean.class);
+        if (isAsset != null) {
+            extensions.append("|  IS_AUGMENTATION: | _").append(isAug).append("_ |");
+            extensions.append(System.getProperty("line.separator"));
+        }
+        SimpleUri defaultWorldGenerator = StandardModuleExtension.getDefaultWorldGenerator(module);
+        if (defaultWorldGenerator != null && defaultWorldGenerator.isValid()) {
+            extensions.append("|  Default World Generator: | _").append(defaultWorldGenerator).append("_ |");
+            extensions.append(System.getProperty("line.separator"));
+        }
+
+        StringBuilder out = new StringBuilder();
+        out.append(System.getProperty("line.separator"));
+        if (extensions.length() > 0) {
+            out.append(System.getProperty("line.separator"));
+            out.append("**Extensions:**");
+            out.append(System.getProperty("line.separator"));
+            out.append("| Extension | Value |");
+            out.append(System.getProperty("line.separator"));
+            out.append("| -------- | ---------- |");
+            out.append(System.getProperty("line.separator"));
+            out.append(extensions);
+        } else {
+            out.append(System.getProperty("line.separator"));
+            out.append("**Extensions:** _None_");
+        }
+        return out.toString();
+    }
+
+    private static String getOriginModuleUrl(Module module) {
```

Comment:
If you want to go for a bit more type safety you could encode the Url as actual `Url`. This is probably not needed when the main goal is to produce markdown output, though.

Reply:
encoded